hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/ClosePipelineCommandHandler.java

```diff
@@ -122,19 +125,37 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
                   } catch (GroupMismatchException ae) {
                     // ignore silently since this means that the group has been closed by earlier close pipeline
                     // command in another datanode
+                    LOG.debug("Failed to remove group {} for pipeline {} on peer {} since the group has " +
+                        "been removed by earlier close pipeline command handled in another datanode", raftGroupId,
+                        pipelineID, peer.getId());
                   } catch (IOException ioe) {
-                    LOG.warn("Failed to remove group {} for peer {}", raftGroupId, peer.getId(), ioe);
+                    LOG.warn("Failed to remove group {} of pipeline {} on peer {}",
+                        raftGroupId, pipelineID, peer.getId(), ioe);
                   }
                 });
           }
+          // Remove the Ratis group from the current datanode pipeline, might throw GroupMismatchException as
+          // well. It is a no-op for XceiverServerSpi implementations (e.g. XceiverServerGrpc)
+          server.removeGroup(pipelineIdProto);
           LOG.info("Close Pipeline {} command on datanode {}.", pipelineID,
               dn.getUuidString());
         } else {
-          LOG.debug("Ignoring close pipeline command for pipeline {} " +
-              "as it does not exist", pipelineID);
+          LOG.debug("Ignoring close pipeline command for pipeline {} on datanode {} " +
+              "as it does not exist", pipelineID, dn.getUuidString());
         }
+      } catch (GroupMismatchException gme) {
+        // ignore silently since this means that the group has been closed by earlier close pipeline
+        // command in another datanode
+        LOG.debug("The group for pipeline {} on datanode {} has been removed by earlier close " +
+            "pipeline command handled in another datanode", pipelineID, dn.getUuidString());
```

Comment:
@ivandika3 , Since `HddsClientUtils.containsException` below will cover this case, let's remove `catch (GroupMismatchException gme) {...}`?

Reply:
Thank you for the review. Updated.